src/main/java/de/tum/in/www1/artemis/web/rest/metis/conversation/ConversationResource.java

```diff
@@ -107,6 +107,23 @@ public ResponseEntity<Void> switchHiddenStatus(@PathVariable Long courseId, @Pat
         return ResponseEntity.ok().build();
     }
 
+    /**
+     * GET /api/courses/:courseId/unread-messages : Checks for unread messages of the current user
+     *
+     * @param courseId the id of the course
+     * @return ResponseEntity with status 200 (Ok) and the information if the user has unread messages
+     */
+    @GetMapping("/{courseId}/unread-messages")
+    @PreAuthorize("hasRole('USER')")
+    public ResponseEntity<Boolean> switchHiddenStatus(@PathVariable Long courseId) {
+        checkMessagingEnabledElseThrow(courseId);
+
+        var requestingUser = userRepository.getUserWithGroupsAndAuthorities();
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, courseRepository.findByIdElseThrow(courseId), requestingUser);
+        var conversations = conversationService.getConversationsOfUser(courseId, requestingUser);
```

Comment:
```suggestion
        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, courseRepository.findByIdElseThrow(courseId), null);
        var requestingUser = userRepository.getUserWithGroupsAndAuthorities();
        var conversations = conversationService.getConversationsOfUser(courseId, requestingUser);
```
Even if we already have access to this user, I'd prefer to use the null value whenever possible. To reduce interferences.

Reply:
This is done in many other places as well and leaving it out would decrease the performance. Since this is a call that every user executes when loading the site, I especially see the performance more important

Reply:
I hear you. Makes sense.